sonic-xcvrd/scripts/xcvrd

```diff
@@ -672,36 +710,151 @@ class sfp_state_update_task:
+                        next_state = STATE_NORMAL
+                    # this is the originally logic that handled the transceiver change event
+                    # this can be reached in two cases:
+                    #   1. the state has been normal before got the event
+                    #   2. the state was init and is transmitted to normal after got the event.
+                    #      this is for the vendors who don't implement "system_not_ready/system_becom_ready" logic
+                    for key, value in port_dict.iteritems():
+                        logical_port_list = platform_sfputil.get_physical_to_logical(int(key))
+                        for logical_port in logical_port_list:
+                            if value == SFP_STATUS_INSERTED:
+                                logger.log_info("Got SFP inserted event")
+                                rc = post_port_sfp_info_to_db(logical_port, int_tbl, transceiver_dict)
+                                # If we didn't get the sfp info, assuming the eeprom is not ready, give a try again.
+                                if rc == SFP_EEPROM_NOT_READY:
+                                    logger.log_warning("SFP EEPROM is not ready. One more try...")
+                                    time.sleep(TIME_FOR_SFP_READY_SECS)
+                                    post_port_sfp_info_to_db(logical_port, int_tbl, transceiver_dict)
+                                post_port_dom_info_to_db(logical_port, dom_tbl)
+                                post_port_dom_threshold_info_to_db(logical_port, dom_tbl)
+                                notify_media_setting(logical_port, transceiver_dict, app_port_tbl)
+                                transceiver_dict.clear()
+                            elif value == SFP_STATUS_REMOVED:
+                                logger.log_info("Got SFP removed event")
+                                del_port_sfp_dom_info_from_db(logical_port, int_tbl, dom_tbl)
+                            else:
+                                # TODO, SFP return error code, need handle accordingly.
+                                logger.log_warning("Got unknown event {}, ignored".format(value))
+                                continue
+                else:
+                    next_state = STATE_EXIT
+            elif event == SYSTEM_FAIL:
+                # no matter which state current it is, it's fatal
+                next_state = STATE_EXIT
+                logger.log_error("Got system_fail event on state {}, exiting".format(state))
             else:
-                # If get_transceiver_change_event() return error, will clean up the DB and then exit
-                # TODO: next step need to define more error types to handle accordingly.
-                logger.log_error("Failed to get transceiver change event. Exiting...")
+                logger.log_warning("Got unknown event {} on state {}.".format(event, state))
+
+            if not next_state == state:
```

Comment:
>if not next_state == state: [](start = 12, length = 27)

For readability, use `if next_state != state`

Reply:
fixed.